tahoma_api/client.py

```diff
@@ -115,53 +97,98 @@ async def fetch_event_listener(self, listener_id: str) -> List[Any]:
         Per-session rate-limit : 1 calls per 1 SECONDS period for this particular
         operation (polling)
         """
-        response = await self.__make_http_request("POST", f"events/{listener_id}/fetch")
+        response = await self.__do_http_request("POST", f"events/{listener_id}/fetch")
 
         return response
 
-    # async def execute_action_group(
-    #     self, actions: List[Command], label: str = "python-tahoma-api"
-    # ) -> List[Any]:
-    #     """ Execute a non-persistent action group.
-    #     The executed action group does not have to be persisted on the server before
-    #     use.
-    #     Per-session rate-limit : 50 calls per 24 HOURS period for all operations of the
-    #     same category (exec)
-    #     """
-    #     payload = {"label": label, "actions": actions}
-    #     response = await self.__make_http_request("POST", "exec/apply", payload)
-    #
-    #     return response
-
-    async def __make_http_request(
+    async def execute_action_group(
+        self,
+        actions: List[Command],
+        label: str = "python-tahoma-api",
+        mode: Optional[CommandMode] = None,
+    ) -> List[Any]:
+        """ Execute a non-persistent action group.
+        The executed action group does not have to be persisted on the server before
+        use.
+        Per-session rate-limit : 50 calls per 24 HOURS period for all operations of the
+        same category (exec)
+        """
+        payload = {"label": label, "actions": actions}
+        supported_modes = ["geolocated", "highPriority", "internal"]
+        endpoint = "exec/apply"
+
+        # TODO Change mode / supported_modes to ENUM
+        if mode in supported_modes:
+            endpoint = f"{endpoint}/{mode}"
+
+        response = await self.__do_http_request("POST", endpoint, payload)
+
+        return response
+
+    async def get_current_execution(self, exec_id: str) -> List[Any]:
+        """ Get an action group execution currently running """
+        response = await self.__do_http_request("GET", f"/exec/current/{exec_id}")
+        # TODO Strongly type executions
+
+        return response
+
+    async def get_current_executions(self) -> List[Any]:
+        """ Get all action groups executions currently running """
+        response = await self.__do_http_request("GET", "/exec/current")
+        # TODO Strongly type executions
+
+        return response
+
+    async def __do_http_request(
```

Comment:
In a future PR, it would be nice to create dedicated get and post methods.

Reply:
@tetienne agree, I am not a fan of this huge method yet.